examples/wandb_sweeps/sweep.py

```diff
@@ -29,22 +29,48 @@ def train_sweep(
         argv: command-line arguments.
     """
     populate_config(config, temp_config)
-    run_sweep(argv)
+    exit_code = run_sweep(argv)
+    wandb.finish(exit_code=exit_code)
 
 
-def run_sweep(argv: List[str]) -> None:
+def run_sweep(argv: List[str]) -> int:
     """Actually runs the sweep.
 
     Args:
         argv: command-line arguments.
 
+    Returns:
+        int: The exit code from the subprocess. 0 for success.
+
     We encapsulate each run by using a separate subprocess, which ought to
     ensure that memory is returned (etc.).
     """
     try:
-        subprocess.check_call(argv)
-    except subprocess.CalledProcessError as error:
-        logging.error("Subprocess error: %s", error)
+        process = subprocess.Popen(argv, stderr=subprocess.PIPE)
```

Comment:
The try/except part is no longer necessary. The outer call is wrapped in a try/except.